From 01a455361635a4eb495b0a9f4ae656617c00bfd4 Mon Sep 17 00:00:00 2001
From: Sampath Vangaveti <sampathv@codeaurora.org>
Date: Fri, 23 Feb 2018 21:22:32 +0530
Subject: [PATCH 18/20] Camera2Client: Fix issue with supported scene modes.

In QTI Camera2Client,
while advertising the supported scene modes for API1,
do not remove scene modes if disabled is supported from HAL3.

Change-Id: Iddf649ddb551a34a8510a163673a864b4e785c69
---
 services/camera/libcameraservice/api1/qticlient2/Parameters.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp b/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
index 08bb0e227..48517f078 100644
--- a/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
+++ b/services/camera/libcameraservice/api1/qticlient2/Parameters.cpp
@@ -554,7 +554,7 @@ status_t Parameters::initialize(const CameraMetadata *info, int deviceVersion,
             addComma = true;
             switch (availableSceneModes.data.u8[i]) {
                 case ANDROID_CONTROL_SCENE_MODE_DISABLED:
-                    noSceneModes = true;
+                    addComma = false;
                     break;
                 case ANDROID_CONTROL_SCENE_MODE_FACE_PRIORITY:
                     // Not in old API
-- 
2.17.1

